test/expire_cache_test.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class ExpireCacheTest < IdentityCache::TestCase
+  def setup
+    super
+
+    Item.cache_index(:title, unique: true)
+    Item.cache_index(:id, :title, unique: true)
+
+    @record = Item.new
+    @record.id = 1
+    @record.title = "bob"
+
+    Spy.on(Item.cached_primary_index, :load_one_from_db).and_return do
+      @record
+    end
+  end
+
+  def test_expire_cache_hit
+    assert_equal(@record, Item.fetch(1))
+    refute_equal(IdentityCache::DELETED, backend.read(@record.primary_cache_index_key))
+    assert(@record.expire_cache)
+    assert_equal(IdentityCache::DELETED, backend.read(@record.primary_cache_index_key))
+  end
+
+  def test_expire_cache_on_failure
+    assert_equal(@record, Item.fetch(1))
+    refute_equal(IdentityCache::DELETED, backend.read(@record.primary_cache_index_key))
+
+    Spy.on(backend, :write_entry).and_return { false }
```

Comment:
`spy` isn't a widely used gem at Shopify, so I would prefer to limit usage on it to where integration testing or mocha aren't suitable.  In this case, it isn't even being used to spy on a method call without interfering with the call, it is just stubbing the call, which mocha could be used for.

Although, even better would be to avoid any fake to give confidence that this works as expected, which could be done by using a cache adapter configured with an open port.  I've done that in the past at Shopify with the following helper method

```ruby
module RandomOpenPort
  def self.find
    socket = Socket.new(:INET, :STREAM)
    socket.bind(Addrinfo.tcp("127.0.0.1", 0))
    socket.local_address.ip_port
  ensure
    socket.close if socket
  end
end
```

Reply:
Good point.  That seems like a good reason to use mocking.  I'd mostly just prefer using mocha over spy for stubbing.  A mocked server could work, but it might be a heavier weight solution than what is needed to test the short circuiting logic.